test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,373 @@
+
+	seedCreationConfig = newCfg
+
+	return seedCreationConfig
+}
+
+func (f *SeedRegistrationFramework) CreateSeed(ctx context.Context) error {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+
+	var (
+		seed    = &gardencorev1beta1.Seed{}
+		err     error
+		ingress = "ingress"
+	)
+
+	refShoot := gardencorev1beta1.Shoot{}
+
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.ShootedSeedName), &refShoot); err != nil {
+		return err
+	}
+
+	seed.Name = f.Config.SeedName
+	ingressDommain := fmt.Sprintf("%s.%s", ingress, *refShoot.Spec.DNS.Domain)
+	seed.Spec.DNS = gardencorev1beta1.SeedDNS{
+		IngressDomain: ingressDommain,
+	}
+	seed.Spec.SecretRef = &corev1.SecretReference{
+		Name:      f.Config.SeedSecretName,
+		Namespace: f.Config.SeedSecretNamespace,
+	}
+
+	seed.Labels = make(map[string]string)
+	seed.Labels["gardener.cloud/role"] = "seed"
+
+	seed.Spec.Backup = &gardencorev1beta1.SeedBackup{}
+	seed.Spec.Backup.SecretRef = corev1.SecretReference{}
+	seed.Spec.Backup.SecretRef.Name = f.Config.BackupSecretName
+	seed.Spec.Backup.SecretRef.Namespace = f.Config.BackupSecretNamespace
+	seed.Spec.Backup.Provider = f.Config.BackupSecretProvider
+
+	seed.Spec.Provider.Region = f.Config.Provider
+	seed.Spec.Provider.Type = f.Config.ProviderType
+
+	seed.Spec.Networks = gardencorev1beta1.SeedNetworks{}
+	seed.Spec.Networks.BlockCIDRs = strings.Split(f.Config.NetworingServices, ",")
+	seed.Spec.Networks.Nodes = &f.Config.NetworingkNodes
+	seed.Spec.Networks.Pods = f.Config.NetworingPods
+	seed.Spec.Networks.Services = f.Config.NetworingServices
+	podsDef := f.Config.NetworingDefPods
+	seed.Spec.Networks.ShootDefaults = &gardencorev1beta1.ShootNetworks{}
+	seed.Spec.Networks.ShootDefaults.Pods = &podsDef
+	servicesDef := f.Config.NetworkingDefServices
+	seed.Spec.Networks.ShootDefaults.Services = &servicesDef
+	seed.Spec.Volume = &gardencorev1beta1.SeedVolume{
+		MinimumSize: resource.NewScaledQuantity(5, resource.Giga),
+	}
+	seed.Spec.Settings = &gardencorev1beta1.SeedSettings{
+		Scheduling: &gardencorev1beta1.SeedSettingScheduling{
+			Visible: false,
+		},
+	}
+
+	f.Seed = seed
+	if err = f.GenerateSeedSecret(ctx); err != nil {
+		f.Logger.Fatalf("Cannot generate seed secret %s", err)
+		return err
+	}
+
+	f.Logger.Infof("Creating seed secret %s in namespace %s", f.Secret.GetName(), f.Secret.GetNamespace())
+	//Apply secret to the cluster
+	if err = f.createSeedSecret(ctx, f.Secret); err != nil {
+		f.Logger.Fatalf("Cannot seed create secret %s: %s", f.Secret.GetName(), err)
+		return err
+	}
+	f.Logger.Infof("Registering seed %s", seed.GetName())
+	if err := PrettyPrintObject(f.Secret); err != nil {
+		f.Logger.Fatalf("Cannot decode seed %s: %s", seed.GetName(), err)
+		return err
+	}
+	return f.GardenerFramework.CreateSeed(ctx, f.Seed)
+}
+
+func (f *SeedRegistrationFramework) createSeedSecret(ctx context.Context, secret *corev1.Secret) error {
```

Comment:
We don't need this method tbh. You can just call f.GardenClient.DirectClient().Create() and handle the already exists error in the calling code as above.